client/src/components/Chat.vue

```diff
@@ -0,0 +1,197 @@
+                <div class="col-9">
+                    <v-text-field
+                        label="Message"
+                        outlined
+                        clearable
+                        v-model="newMessage"
+                    > </v-text-field>
+                </div>
+                <div class="col-3">
+                    <v-card-actions>
+                        <v-btn id="send-btn" color="blue darken-1" text @click.native="sendMessage">Send</v-btn>
+                    </v-card-actions>
+                </div>
+            </div>
+            <div v-else class="text-center">
+                <p> {{ info }} </p>
+            </div>
+        </div>
+    </v-card>
+</div>
+</template>
+
+<script> 
+import unoService from '../services/unoService';
+
+export default {
+    name: "Chat",
+    components: {
+
+    },
+    props: ['gameState'],
+    data() {
+        return {
+            newMessage: '',
+            messages: [],
+
+            info: null,
+            displayInfo: false,
+            
+            players: [],
+            // It would be best to know a max number of people able to play
+            // TODO: get ride of the blue option similar to the `from-me`
+            messageColors: ['primary', 'success', 'warning', 'indigo', 'purple', 'pink', 'red', 'orange', 'yellow', 'green', 'teal', 'cyan'],
+
+            loop_scroll: true,
+        }
+    },
+    watch: {
+        gameState(newVal, oldVal) {
+
+            this.messages = newVal.messages;
+            this.players = newVal.all_players;
+
+            // There are .length errors if the messages is null
+            if (this.messages != null) {
+                // Assign Message Colors to the players
+                for (var i = 0; i < this.players.length; i++) {
+                    for (var j = 0; j < this.messages.length; j++) {
+                        this.messages[j].player.color = this.messageColors[i]
+                    }
+                }
+
+                // if I don't need to scroll .. Don't
+                if (newVal.messages.length > 5) {
+                    // If we have a new message scroll down
+                    var len__new = newVal.messages.length - 1;
+                    var len__old = oldVal.messages.length - 1;
+                    if ( len__new !== len__old ) {
+                        this.loop_scroll = true;
+                    }
+                }
+            }
+            this.info = null;
+            this.displayInfo = false;
+        }
+    },
+    methods: {
+        scroll() {
+            var div = document.getElementById('chatcard');
+            div.scrollTop = div.scrollHeight - div.clientHeight;
+        },
+        async sendMessage() {
+            if (this.newMessage != "") {
+                this.info = 'Sending to Server';
+                this.displayInfo = true;
+                let res = await unoService.sendMessage(this.$route.params.id, this.gameState.current_player.id, this.newMessage);
```

Comment:
Thank you for using the unoService here :)  I love seeing this being used.

Reply:
It's a super cool way of using axios and I'll definitely be implementing it in my future projects